### DISCUSSION TITLE
Is it possible to extend included presets?

### DISCUSSION THREAD

**Discussion by ablekhman** [renovatebot/renovate#8407] [Request Help] [answered]
I am wondering if it is possible to extend included presets. For example, there is included `group:springCore` preset:
```json
{
  "packageRules": [
    {
      "groupName": "spring core",
      "packagePatterns": [
        "^org.springframework:"
      ]
    }
  ]
}
```
I would like to add more rules to this preset. For example, I want to restrict Spring Framework updates to patch versions only:
```json
{
  "packageRules": [
    {
      "extends": ["group:springCore"],
      "updateTypes": ["patch"]
    }
  ]
}
```
Is extending included presets supported?

**Comment by viceice** [comment]
Should work 🤗

**Comment by rarkins** [comment, marked as answer]
You *can* extend presets, but the example you gave will not work. If you want patch-only in the example you gave, it needs to look like this:

```
{
  "packageRules": [
    {
      "extends": ["monorepo:springCore"],
      "updateTypes": ["minor", "major"],
      "enabled": false
    }
  ]
}
```

i.e. match the spring core mono repo, match update types minor and major, and then disable them. `updateTypes` is a *matching* clause, not an *instruction* clause.

**Reply by ablekhman** [reply]
Thank you!

P.S. There is no `monorepo:springCore` preset. Did you mean `group:springCore`?

**Reply by rarkins** [reply]
Yes, use that. We usually have a `monorepo:x` for every `group:x` but seem to have missed it for that one.
